container-engine-lib/lib/backend_impls/remote_context_backend/remote_context_kurtosis_backend.go

```diff
@@ -0,0 +1,208 @@
+package remote_context_backend
+
+import (
+	"context"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/api_container"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/enclave"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/engine"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/exec_result"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_collector"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_database"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/networking_sidecar"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/stacktrace"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"net"
+)
+
+// RemoteContextKurtosisBackend is a dual context holding a reference to a local backend running on Docker (k8s is
+// currently not supported to work with a remote Kurtosis backend) and a reference to a remote Kurtosis backend (which
+// currently is a Docker backend, we do not support anything else).
+// Basically, requests are dispatched among those 2 backends depending on which area of Kurtosis they touch.
+// This might change, but right now the engine runs locally, therefore engine interacting endpoint goes to the local
+// backend, and all the rest goes to the remote backend, as all other containers (including API container) should be
+// running in the remote backend
+type RemoteContextKurtosisBackend struct {
+	localKurtosisBackend  backend_interface.KurtosisBackend
+	remoteKurtosisBackend backend_interface.KurtosisBackend
+}
+
+// newRemoteContextKurtosisBackend instantiate a new RemoteContextKurtosisBackend but is private. Use the corresponding
+// backend_creator to build such a backend
+func newRemoteContextKurtosisBackend(
+	localBackend backend_interface.KurtosisBackend,
+	contextBackend backend_interface.KurtosisBackend,
+) *RemoteContextKurtosisBackend {
+	return &RemoteContextKurtosisBackend{
+		localKurtosisBackend:  localBackend,
+		remoteKurtosisBackend: contextBackend,
+	}
+}
+
+func (backend *RemoteContextKurtosisBackend) FetchImage(ctx context.Context, image string) error {
+	// without knowing which backend will need the image, it's tricky to filter which one should fetch it.
```

Comment:
could it be simpler if we check if the image is "kurtosis/engine", then its engine otherwise its APIC.

Reply:
This would be a huge hack, not sure we necessarily want to do it here, but yes we could. I'd bias toward keeping it generic here, at the cost of downloading the images on both client, but I could be convinced otherwise

Reply:
It just seems more simpler and less compute. In the future if we want to support multiple images we could handle that. Like if we need to fetch an image in the APIC but not in the engine we would now fail if it fails to fetch on Engine which it shouldn't IMO?

Reply:
It's just that we're introducing a behaviour that is not extensible, therefore making future change to this effectively a regression. But fair enough, I do that with a big comment